test/integration-test-main.sh

```diff
@@ -258,7 +261,10 @@ function test_remove_nonempty_directory {
     echo "Testing removing a non-empty directory"
     mk_test_dir
     touch "${TEST_DIR}/file"
-    rmdir "${TEST_DIR}" 2>&1 | grep -q "Directory not empty"
+    if ! (rmdir "${TEST_DIR}" 2>&1 | grep -q "Directory not empty"); then
+        echo "Did not get \"Directory not empty\""
+        exit 1
+    fi
```

Comment:
What is the intent of this change?  Previously we exit if we do not see "Directory not empty" and pass only if we do see it.